src/dart/lib/widgets.dart

```diff
@@ -0,0 +1,288 @@
+  /// How to clip the image.
+  ///
+  /// This can be used to enable anti-aliased clips if the image is rotated, or
+  /// disable clips entirely (for improved performance) if the image does not
+  /// need them.
+  final Clip clipBehavior;
+
+  @override
+  State<IconVGImage> createState() => _IconVGImageState();
+}
+
+class _IconVGImageState extends State<IconVGImage> {
+  AssetBundle? _assetBundle;
+  Uint8List? _bytes;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final AssetBundle assetBundle = DefaultAssetBundle.of(context);
+    if (assetBundle != _assetBundle) {
+      _assetBundle = assetBundle;
+      _bytes = null;
+      _fetchBytes();
+    }
+  }
+
+  @override
+  void didUpdateWidget(IconVGImage oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.asset != widget.asset) {
+      _bytes = null;
+      _fetchBytes();
+    }
+  }
+
+  Future<void> _fetchBytes() async {
+    assert(_assetBundle != null);
+    final ByteData data = await _assetBundle!.load(widget.asset);
+    if (mounted) {
+      setState(() {
+        _bytes = data.buffer.asUint8List();
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return RawIconVGImage(
+      bytes: _bytes,
+      palette: widget.palette,
+      size: widget.size,
+      clipBehavior: widget.clipBehavior,
+    );
+  }
+}
+
+/// Render the specified [bytes] as an IconVG image.
+///
+/// If the image is an asset, consider using [IconVGImage] instead.
+class RawIconVGImage extends StatefulWidget {
+  const RawIconVGImage({
+    Key? key,
+    this.bytes,
+    this.palette,
+    this.size,
+    this.clipBehavior = Clip.hardEdge,
+  }) : super(key: key);
+
+  /// The bytes. Must be in the IconVG format.
+  final Uint8List? bytes;
+
+  /// The custom color palette used to decode the image.
+  ///
+  /// Must not have more than 64 colors.
+  final List<Color>? palette;
+
+  /// The target render size. Defaults to the [BoxConstraints.maxHeight].
+  final Size? size;
+
+  /// How to clip the image.
+  ///
+  /// This can be used to enable anti-aliased clips if the image is rotated, or
+  /// disable clips entirely (for improved performance) if the image does not
+  /// need them.
+  final Clip clipBehavior;
+
+  @override
+  State<RawIconVGImage> createState() => _RawIconVGImageState();
+}
+
+class _RawIconVGImageState extends State<RawIconVGImage> {
+  IconVGFile? _image;
+  double? _decodedHeight;
+
+  @override
+  void didUpdateWidget(RawIconVGImage oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.bytes != widget.bytes ||
+        oldWidget.palette != widget.palette ||
+        oldWidget.clipBehavior != widget.clipBehavior) {
+        // widget.size is handled in build
+      _image = null;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.bytes == null) {
+      return SizedBox.fromSize(size: widget.size ?? Size.zero);
+    }
+    return LayoutBuilder(
```

Comment:
It would be good to avoid using a LayoutBuilder here - maybe just some way to quickly parse the size of the IconVG before/while parsing the rest of the file would work better.

Reply:
The actual rendered height is what is being obtained here. IconVGs have a level-of-detail feature that allows them to render differently based on the rasterization size.

Reply:
Yes, but it's being obtained after doing a layout pass because of the layout builder - or am I missing something?

Reply:
Not sure I follow. LayoutBuilder builds during layout, it's still single-pass.

Reply:
Ok - this should be fine then.